cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -178,8 +178,14 @@ func generateMountsModeRecover(mst *initramfsMountsState, recoverySystem string)
 		return err
 	}
 	if !isRecoverDataMounted {
-		// TODO:UC20: data may need to be unlocked
-		fmt.Fprintf(stdout, "/dev/disk/by-label/ubuntu-data %s\n", boot.InitramfsHostUbuntuDataDir)
+		// TODO: do we do this in recover mode too ?
+		const lockKeysForLast = true
+		device, err := unlockIfEncrypted("ubuntu-data", lockKeysForLast)
```

Comment:
currently this will _not_ measure the model on ubuntu-boot, should we still do that? this code path for recover also lacks this bit of code from the codepath for run mode:

https://github.com/snapcore/snapd/blob/d0956199578dd31578b0cd84d5afb40bbee477fa/cmd/snap-bootstrap/cmd_initramfs_mounts.go#L344-L356

Reply:
we actually need the same code as install and measure the model from the recovery system

Reply:
this is already done by the shared helper between install and recover modes